runtime/include/module.h

```diff
@@ -141,12 +158,62 @@ module_release(struct module *module)
 	return;
 }
 
+static inline struct wasm_stack *
+module_allocate_stack(struct module *self)
+{
+	assert(self != NULL);
+
+	struct wasm_stack *stack = wasm_stack_pool_remove_nolock(&self->pools.stack[worker_thread_idx]);
+
+	if (stack == NULL) {
+		stack = wasm_stack_new(self->stack_size);
+		if (unlikely(stack == NULL)) return NULL;
+	}
+
+	return stack;
+}
+
+static inline void
+module_free_stack(struct module *self, struct wasm_stack *stack)
+{
+	wasm_stack_reinit(stack);
+	wasm_stack_pool_add_nolock(&self->pools.stack[worker_thread_idx], stack);
+}
+
+static inline struct wasm_memory *
+module_allocate_linear_memory(struct module *module)
+{
+	assert(module != NULL);
+
+
+	size_t initial = (size_t)module->abi.starting_pages * WASM_PAGE_SIZE;
+	size_t max     = (size_t)module->abi.max_pages * WASM_PAGE_SIZE;
+
+	assert(initial <= (size_t)UINT32_MAX + 1);
+	assert(max <= (size_t)UINT32_MAX + 1);
+
+	struct wasm_memory *linear_memory = wasm_memory_pool_remove_nolock(&module->pools.memory[worker_thread_idx]);
+	if (linear_memory == NULL) {
+		linear_memory = wasm_memory_new(initial, max);
```

Comment:
`new` isn't really a thing in C. `_alloc` is usually "allocate and initialize", and `_init` is "initialize a piece of memory". alloc calls init. Init exists as a separate function for stack and global data.